src/engine/threaded_engine_perdevice.cc

```diff
@@ -87,21 +92,34 @@ class ThreadedEnginePerDevice : public ThreadedEngine {
         int nthread = gpu_worker_nthreads_;
         int dev_id = ctx.dev_id;
         if (is_copy) {
+          auto ptr =
```

Comment:
My understanding is that this is called a finite number of times based upon the number of devices (ie once for execution queue init, once for copy queue init)